iceoryx_posh/test/integrationtests/test_service_discovery.cpp

```diff
@@ -580,9 +589,12 @@ TEST_F(ServiceDiscovery_test, ServiceDiscoveryIsNotifiedByListenerAboutSingleSer
     const iox::capro::ServiceDescription SERVICE_DESCRIPTION("Moep", "Fluepp", "Shoezzel");
     iox::popo::UntypedPublisher publisher(SERVICE_DESCRIPTION);
 
-    this->InterOpWait();
+    while (!callbackWasCalled.load())
+    {
+        std::this_thread::yield();
+    }
 
-    EXPECT_TRUE(callbackWasCalled);
+    EXPECT_TRUE(callbackWasCalled.load());
```

Comment:
We would not have made it out of the loop otherwise I think but ok. (So in theory we could call an unconditional `SUCCEED();`.

Reply:
See your point. However, my eye always searches for an `EXPECT_*` at the end of a test case, so I would like to leave it here in this verbose way, if you don't mind.

Reply:
It`s fine, just saying as I am hardwired to avoid computation if the result is already determined.